src/io/GeantModel.hh

```diff
@@ -0,0 +1,111 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantModel.hh
+//! \brief Enumerator for the available physics models
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * This enum does not exists in Geant4. It was created to safely access the
+ * many physics tables imported from it.
+ * Geant4 has model names hardcoded as strings. This class provides an
+ * interface to select an enum from a string, while still working as an enum:
+ *
+ * \code
+ *  GeantModel model("KleinNishina")
+ *  if (model.get() == GeantModel::KleinNishina) { ... }
+ * \endcode
+ *
+ * FOR REVIEW: The enum keeps the Geant4 names, not following Celeritas code
+ * standards.
+ */
+class GeantModel
+{
+  public:
+    enum Value
+    {
+        // EM model list
+        not_defined,
+        BraggIon,
+        BetheBloch,
+        UrbanMsc,
+        ICRU73QO,
+        WentzelVIUni,
+        hBrem,
+        hPairProd,
+        eCoulombScattering,
+        Bragg,
+        MollerBhabha,
+        eBremSB,
+        eBremLPM,
+        LivermorePhElectric,
+        KleinNishina,
+        BetheHeitlerLPM,
+        LivermoreRayleigh,
+        MuBetheBloch,
+        MuBrem,
+        muPairProd
+    };
+
+  public:
+    // Constructors
+    GeantModel() : value_(Value::not_defined) {}
+    GeantModel(std::string model_name)
+    {
+        GeantModel model;
+        model(model_name);
+        this->value_ = model.value_;
+    }
+    GeantModel(Value model) : value_(model) {}
+
+    // Default destructor
+    ~GeantModel() = default;
+
+    // Fetch enum value
+    Value get() { return this->value_; }
+
+    // >>> OPERATORS
+         operator Value() const { return value_; }
+    void operator()(std::string model_name)
```

Comment:
With this `operator()` and implicit `operator Value` your class is trying to be multiple things at once. I would make things simpler by making an `enum class GeantModel` (and omit the `not_defined` option)  and define a helper function `GeantModel name_to_model(const std::string& g4name);` in a CC file:
```c++
GeantModel name_to_model(const std::string& g4name)
{
    const static std::unordered_map<std::string, GeantModel> model_map = {
        {"BraggIon", GeatModel::bragg_ion},
        {"BetheBloch", GeatModel::bethe_bloch},
        //...
    };
    auto iter = model_map.find(g4name);
    CHECK(iter != model_map.end());
    return iter->second;
}
```
I think for your enum, since it's sitting inside `src/`, it should follow celeritas naming conventions.

In fact, it's possible we'll have to use this model enum in the CPU/GPU part of the code (as a form of static polymorphism), so that's another reason for having well-behaved names and putting the string translation in a different function.